components/brave_rewards/core/common/signer.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/core/common/signer.h"
+
+#include <openssl/evp.h>
+#include <openssl/hkdf.h>
+#include <utility>
+
+#include "crypto/random.h"
+#include "tweetnacl.h"  // NOLINT
+
+namespace {
+
+const uint8_t kHkdfSalt[] = {
+    126, 244, 99,  158, 51,  68,  253, 80,  133, 183, 51,  180, 77,
+    62,  74,  252, 62,  106, 96,  125, 241, 110, 134, 87,  190, 208,
+    158, 84,  125, 69,  246, 207, 162, 247, 107, 172, 37,  34,  53,
+    246, 105, 20,  215, 5,   248, 154, 179, 191, 46,  17,  6,   72,
+    210, 91,  10,  169, 145, 248, 22,  147, 117, 24,  105, 12};
+
+const int kSeedLength = 32;
+
+const int kSaltLength = 64;
+
+}  // namespace
+
+namespace brave_rewards::internal {
+
+Signer::Signer(std::vector<uint8_t> public_key, std::vector<uint8_t> secret_key)
```

Comment:
When do we expect the private keys and seed to be cleaned up in memory?

Reply:
`Signer` and `RequestSigner` are short-lived (all instances are function-local currently). The recovery seed is typically read from the Preferences file where it is stored.

Reply:
Ok, let's do this as a follow up issue to improve things here but we should ideally be clearing from memory as soon as we no longer expect it to be used. To do this [using something like](https://github.com/brave/brave-core/blob/9824e1e55d742225af5de719b9a9693bf8595e89/components/brave_wallet/browser/internal/hd_key.h#L24C43-L24C62) `SecureVector` is a good way to do this.

Here's an example if you want to take a look at how we do this elsewhere: https://github.com/brave/brave-core/blob/9824e1e55d742225af5de719b9a9693bf8595e89/components/brave_wallet/browser/internal/hd_key.cc#L52

Reply:
https://github.com/brave/brave-browser/issues/35248